src/sdam/monitor.rs

```diff
@@ -127,9 +127,8 @@ impl HeartbeatMonitor {
             let mut topology_check_requests_subscriber =
                 topology.subscribe_to_topology_check_requests();
 
-            if self.check_server(&topology, &server).await {
-                topology.notify_topology_changed();
-            }
+            self.check_server(&topology, &server).await;
+            topology.notify_topology_changed();
```

Comment:
Once I added a commit to fix RUST-1317, some tests started failing due to this `if` statement. The problem is if the error stayed the same, this check would not call `notify_topology_changed`, which in turn would mean that the operation blocked on server selection would not try again and would not request another immediate check. This means that the blocked operation wouldn't succeed until the next heartbeat regularly scheduled, which could be 10s of seconds in the future.

Note that this new implementation matches the pseudocode in the server monitoring spec as well as the Python implementation. Also, by passing the test added in [DRIVERS-1251](https://jira.mongob.org/browse/DRIVERS-1251), the C#, Go and Java likely have this behavior too. The spec does allow for our prior only-notify-if-changed behavior though, so I filed  [DRIVERS-2329](https://jira.mongob.org/browse/DRIVERS-2329) to require the behavior seen here.